packages/p2p-v2/src/pages/my-profile/screens/MyProfileStats/MyProfileStatsLimitModal.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useEffect } from 'react';
+import Modal from 'react-modal';
+import { useAdvertiserUpdate } from '@deriv/api';
+import { Button } from '@deriv-com/ui/dist/components/Button';
+import { Text } from '@deriv-com/ui/dist/components/Text';
+import './MyProfileStatsLimitModal.scss';
+
+type TMyProfileStatsLimitModalProps = {
+    currency: string;
+    isModalOpen: boolean;
+    onRequestClose: () => void;
+};
+
+type TCustomStyles = {
+    overlay: ReactModal.Styles['overlay'];
+};
+
+const customStyles: TCustomStyles = {
+    overlay: {
+        alignItems: 'center',
+        backgroundColor: 'rgba(0, 0, 0, 0.72)',
+        display: 'flex',
+        justifyContent: 'center',
+        zIndex: 9999,
+    },
+};
+
+const MyProfileStatsLimitModal = ({ currency, isModalOpen, onRequestClose }: TMyProfileStatsLimitModalProps) => {
+    const { data, error, isLoading, isSuccess, mutate } = useAdvertiserUpdate();
+    const { daily_buy_limit, daily_sell_limit } = data ?? {};
+    useEffect(() => {
+        Modal.setAppElement('#v2_modal_root');
+    }, []);
+
+    const getModalContent = () => {
+        //TODO: modal header title to be moved out if needed according to implementation
+        if (isLoading) {
+            //TODO: replace with @deriv/ui loading component
+            return <div className='p2p-v2-my-profile-stats-limit-modal__loader'>loading</div>;
+        } else if (isSuccess) {
+            return (
+                <>
+                    <Text className='p2p-v2-my-profile-stats-limit-modal__title' color='prominent' weight='bold'>
+                        Success!
+                    </Text>
+                    <Text as='p' className='p2p-v2-my-profile-stats-limit-modal__text' color='prominent' size='sm'>
+                        {`Your daily limits have been increased to ${daily_buy_limit} ${currency} (buy) and ${daily_sell_limit} ${currency} (sell).`}
+                    </Text>
+                    <div className='p2p-v2-my-profile-stats-limit-modal__footer'>
+                        <Button onClick={onRequestClose} size='lg' textSize='sm'>
```

Comment:
are you also getting issues for the deriv/ui components related to ` Its type 'FC<TextProps>' is not a valid JSX element type.` if so, just wait a while, as shayan is creating a fix for this soon

Reply:
you can now update with the latest one, he has updated the package 🙏